core/arch/arm/plat-k3/drivers/ti_sci.c

```diff
@@ -364,6 +364,118 @@ int ti_sci_get_dkek(uint8_t sa2ul_instance,
 	return 0;
 }
 
+/**
+ * ti_sci_read_otp_mmr() - Get the Extended OTP
+ * @mmr_idx:		32-bit MMR index
+ * @val:		Value of the 32-bit MMR
+ *
+ * Reads the extended OTP bits from efuse
+ *
+ * Return: 0 if all goes well, else appropriate error message
+ */
+int ti_sci_read_otp_mmr(uint8_t mmr_idx, uint32_t *val)
+{
+	struct ti_sci_msg_req_read_otp_mmr req = { };
+	struct ti_sci_msg_resp_read_otp_mmr resp = { };
+	struct ti_sci_xfer xfer = { };
+	int ret = 0;
+
+	ret = ti_sci_setup_xfer(TI_SCI_MSG_READ_OTP_MMR, 0,
+				&req, sizeof(req), &resp, sizeof(resp), &xfer);
+	if (ret)
+		return ret;
+
+	req.mmr_idx = mmr_idx;
+
+	ret = ti_sci_do_xfer(&xfer);
+	if (ret)
+		return ret;
+
+	*val = resp.mmr_val;
+	memzero_explicit(&resp, sizeof(resp));
```

Comment:
`memzero_explicit(&resp)` also when `ti_sci_do_xfer()` failed?
I'm not shure it's needed, it depends on wheter the transer can failed yet with some responses data in.

Reply:
Hi, there will be some data responses but am not sure if they need to be wiped off as they don't have any sensitive data. @glneo do we need to wipe these off too?

Reply:
Looking into the TI-SCI firmware code, I do not believe there is a way to return error after having successfully sent the response message data, so this shouldn't be an issue. But I also see no harm in clearing out the location just in case.